sql/src/test/java/io/crate/planner/operators/PushDownTest.java

```diff
@@ -359,6 +359,25 @@ public void testWhereClauseIsPushedDownIntoSubRelationOfUnion() {
         );
     }
 
+    @Test
+    public void test_filter_is_pushed_beneath_project_set_if_it_targets_standalone_outputs() {
+        var plan = sqlExecutor.logicalPlan(
+            "SELECT" +
+            "  * " +
+            "FROM " +
+            "  (SELECT x, generate_series(1, x) AS y FROM t1) tt " +
+            "WHERE x > 1 AND y > 2"
+        );
+        var expectedPlan =
+            "RootBoundary[x, y]\n" +
+            "Boundary[x, y]\n" +
+            "Boundary[x, y]\n" +
+            "FetchOrEval[x, generate_series(1, x)]\n" +
+            "ProjectSet[generate_series(1, x) | x]\n" +
+            "Collect[doc.t1 | [x] | ((x > 1) AND (generate_series(1, x) > 2))]\n";
```

Comment:
This is now not a valid plan, `generate_series` cannot be beneath `ProjectSet`

Each part of `queryParts` would have to be validated to see if it can be moved down. And all parts that can be moved down can be joined back to a query using `AndOperator.join` or `AndOperator.of`